api/v1beta1/nutanixfailuredomain_types.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2025 Nutanix
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	// NutanixFailureDomainV2Kind represents the Kind of NutanixFailureDomainV2
+	NutanixFailureDomainV2Kind = "NutanixFailureDomainV2"
+
+	// NutanixFailureDomainV2Finalizer is the finalizer used by the NutanixFailureDomainV2 controller to block
+	// deletion of the NutanixFailureDomainV2 object if there are references to this object in other CRs.
+	NutanixFailureDomainV2Finalizer = "nutanixfailuredomainv2.infrastructure.cluster.x-k8s.io"
+)
+
+// PrismElementFailureDomainSpec defines Prism Element specific failure domain spec.
+type PrismElementFailureDomainSpec struct {
+	// prismElement is to identify the Prism Element in the Prism Central for the failure domain.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="prismElement is immutable once set"
```

Comment:
please add a test for this CEL expression. I did something similar here https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/blob/ebf8cb724171dabc0042c71eb06d04477e20cd1e/main_test.go#L503

Reply:
Thanks. Will add unit tests soon.